lldb/source/Plugins/SymbolFile/NativePDB/UdtRecordCompleter.cpp

```diff
@@ -463,7 +467,14 @@ void UdtRecordCompleter::Record::ConstructRecord() {
       if (iter->second.empty())
         continue;
       parent = iter->second.back();
-      iter->second.pop_back();
+
+      // For structs, if the new fields come after the already added ones
+      // without overlap, go back to the root struct.
+      if (parent != &record && iter->first <= offset &&
+          record.kind == Member::Struct && is_last_end_offset(iter))
+        parent = &record;
+      else
+        iter->second.pop_back();
```

Comment:
```suggestion
      // For structs, if the new fields come after the already added ones
      // without overlap, go back to the root struct.
      if (iter->first <= offset &&
          record.kind == Member::Struct && is_last_end_offset(iter))
        parent = &record;
      else {
        parent = iter->second.back();
        iter->second.pop_back();
      }
```
Maybe move `parent = iter->second.back();` into the else branch.